test/e2e/telemetry_logs_analysis_test.go

```diff
@@ -148,80 +158,99 @@ var _ = Describe(suite.ID(), Label(suite.LabelTelemetryLogsAnalysis), Ordered, f
+			assert.DaemonSetReady(ctx, k8sClient, kitkyma.FluentBitDaemonSet)
+		})
+
+		It("Should have running pipelines", func() {
+			assert.LogPipelineHealthy(ctx, k8sClient, logBackendName)
+			assert.MetricPipelineHealthy(ctx, k8sClient, metricBackendName)
+			assert.TracePipelineHealthy(ctx, k8sClient, traceBackendName)
+
+			assert.LogPipelineHealthy(ctx, k8sClient, otelCollectorLogBackendName)
+			assert.LogPipelineHealthy(ctx, k8sClient, fluentBitLogBackendName)
+			assert.LogPipelineHealthy(ctx, k8sClient, selfMonitorLogBackendName)
 		})
 
 		It("Should push metrics successfully", func() {
-			assert.MetricsFromNamespaceDelivered(proxyClient, metricbackendExportURL, otelCollectorNs, telemetrygen.MetricNames)
+			assert.MetricsFromNamespaceDelivered(proxyClient, metricBackendURL, namespace, telemetrygen.MetricNames)
 		})
 
 		It("Should push traces successfully", func() {
-			assert.TracesFromNamespaceDelivered(proxyClient, tracebackendExportURL, otelCollectorNs)
+			assert.TracesFromNamespaceDelivered(proxyClient, traceBackendURL, namespace)
 		})
 
-		It("Should not have any ERROR/WARNING logs in the OtelCollector containers", func() {
+		It("Should collect logs successfully", func() {
+			assert.LogsDelivered(proxyClient, "", logBackendURL)
+		})
+
+		It("Should collect otel collector component logs successfully", func() {
+			assert.LogsDelivered(proxyClient, "telemetry-", otelCollectorLogBackendURL)
+		})
+
+		It("Should collect fluent-bit component logs successfully", func() {
+			assert.LogsDelivered(proxyClient, "telemetry-", fluentBitLogBackendURL)
+		})
+
+		It("Should collect self-monitor component logs successfully", func() {
+			assert.LogsDelivered(proxyClient, "telemetry-", selfMonitorLogBackendURL)
+		})
+
+		It("Should not have any error/warn logs in the otel collector component containers", func() {
 			Consistently(func(g Gomega) {
-				resp, err := proxyClient.Get(otelCollectorLogbackendExportURL)
+				resp, err := proxyClient.Get(otelCollectorLogBackendURL)
 				g.Expect(err).NotTo(HaveOccurred())
 				g.Expect(resp).To(HaveHTTPStatus(http.StatusOK))
 				g.Expect(resp).To(HaveHTTPBody(
 					Not(ContainLd(ContainLogRecord(SatisfyAll(
 						WithPodName(ContainSubstring("telemetry-")),
-						WithLevel(BeElementOf(errorWarningLevels)),
+						WithLevel(MatchRegexp(logLevelsRegexp)),
 						WithLogBody(Not( // whitelist possible (flaky/expected) errors
 							Or(
 								ContainSubstring("grpc: addrConn.createTransport failed to connect"),
+								ContainSubstring("rpc error: code = Unavailable desc = no healthy upstream"),
+								ContainSubstring("interrupted due to shutdown:"),
+								ContainSubstring("Variable substitution using $VAR will be deprecated"),
 							),
 						)),
 					)))),
 				))
 			}, consistentlyTimeout, periodic.TelemetryInterval).Should(Succeed())
 		})
 
-		AfterAll(func() {
-			format.MaxLength = gomegaMaxLength // restore Gomega truncation
```

Comment:
You need to restore gomega truncation after executing this test suite, since otherwise the other e2e tests might be influenced by it.

Reply:
> **Note:** Gomega truncation was set to unlimited/disabled for this test suite, since some errors or error messages weren't properly caught by the matchers otherwise (Usually when long logs are involved).

```go
format.MaxLength = 0 // remove Gomega truncation
```

Reply:
You are right, sorry